cpp/inc/bond/stream/output_buffer.h

```diff
@@ -246,6 +250,12 @@ class OutputMemoryStream
                 _blobs.push_back(blob(_buffer, _rangeOffset, _rangeSize));
             }
 
+            // cap buffer at 2GB
+            if (_bufferSize >= ((std::numeric_limits<uint32_t>::max)() >> 1))
```

Comment:
Could this be `_bufferSize > (std::numeric_limits<int32_t>::max)()`? Seems like it should be `>`, not `>=`, either way.

Reply:
`>` makes sense. I'd rather leave things in terms of `uint32_t` -- seems more semantically coherent.